packages/wallets/src/phantom/phantom.page.ts

```diff
@@ -0,0 +1,121 @@
+import { WalletConfig } from '../wallets.constants';
+import { WalletPage } from '../wallet.page';
+import expect from 'expect';
+import { test, BrowserContext, Page } from '@playwright/test';
+
+export class PhantomPage implements WalletPage {
+  page: Page | undefined;
+
+  constructor(
+    private browserContext: BrowserContext,
+    private extensionUrl: string,
+    public config: WalletConfig,
+  ) {}
+
+  async navigate() {
+    await test.step('Navigate to phantom', async () => {
+      this.page = await this.browserContext.newPage();
+      await this.page.goto(this.extensionUrl + '/popup.html');
+      await this.page.reload();
+      await this.page.waitForTimeout(1000);
+      await this.closePopover();
+      await this.unlock();
+    });
+  }
+
+  async setup() {
+    await test.step('Setup', async () => {
+      // added explicit route to /onboarding due to unexpected first time route from /popup.html to /onboarding - page is close
+      this.page = await this.browserContext.newPage();
+      await this.page.goto(this.extensionUrl + '/onboarding.html');
+      if (!this.page) throw "Page isn't ready";
+      const firstTime =
+        (await this.page
+          .locator('button:has-text("I already have a wallet")')
+          .count()) > 0;
+      if (firstTime) await this.firstTimeSetup();
+    });
+  }
+
+  async unlock() {
+    await test.step('Unlock', async () => {
+      if (!this.page) throw "Page isn't ready";
+      if ((await this.page.locator('id=password').count()) > 0) {
+        await this.page.fill('input[name=password]', this.config.PASSWORD);
+        await this.page.click('text=Unlock');
+      }
+    });
+  }
+
+  async closePopover() {
+    await test.step('Close popover if exists', async () => {
+      if (!this.page) throw "Page isn't ready";
+      const popover =
+        (await this.page.locator('data-testid=popover-close').count()) > 0;
+      if (popover) await this.page.click('data-testid=popover-close');
+    });
+  }
+
+  async firstTimeSetup() {
+    await test.step('First time setup', async () => {
+      if (!this.page) throw "Page isn't ready";
+      await this.page.click('button:has-text("I already have a wallet")');
+      const inputs = this.page.locator('.sc-bdvvtL >> input');
```

Comment:
looks like a dynamic locator and it may change in the future

Reply:
Fixed